nixos/modules/programs/dconf.nix

```diff
@@ -1,55 +1,174 @@
 { config, lib, pkgs, ... }:
 
-with lib;
-
 let
   cfg = config.programs.dconf;
-  cfgDir = pkgs.symlinkJoin {
-    name = "dconf-system-config";
-    paths = map (x: "${x}/etc/dconf") cfg.packages;
-    postBuild = ''
-      mkdir -p $out/profile
-      mkdir -p $out/db
-    '' + (
-      concatStringsSep "\n" (
-        mapAttrsToList (
-          name: path: ''
-            ln -s ${path} $out/profile/${name}
-          ''
-        ) cfg.profiles
-      )
-    ) + ''
-      ${pkgs.dconf}/bin/dconf update $out/db
-    '';
-  };
+
+  # Compile keyfiles to Dconf DB
+  compileDconfDb = dir: pkgs.runCommand "dconf-db"
+    {
+      nativeBuildInputs = [ pkgs.dconf ];
+    } "dconf compile $out ${dir}";
```

Comment:
Looks like this does not compile ibus database for some reason:

https://github.com/NixOS/nixpkgs/blob/486d2ce7d68af106bb5eccc46bc9f1fd4c7294b4/nixos/modules/i18n/input-method/ibus.nix#L66